docs/sql-ref-syntax-qry-select-groupby.md

```diff
@@ -50,34 +65,28 @@ aggregate_name ( [ DISTINCT ] expression [ , ... ] ) [ FILTER ( WHERE boolean_ex
     is a shorthand for a `UNION ALL` where each leg of the `UNION ALL`
     operator performs aggregation of subset of the columns specified in the `GROUPING SETS` clause.
 
-* **grouping_set**
-
-    A grouping set is specified by zero or more comma-separated expressions in parentheses.
-
-    **Syntax:** `( [ expression [ , ... ] ] )`
-
-* **grouping_expression**
-
-    Specifies the criteria based on which the rows are grouped together. The grouping of rows is performed based on
-    result values of the grouping expressions. A grouping expression may be a column alias, a column position
-    or an expression.
-
 * **ROLLUP**
 
     Specifies multiple levels of aggregations in a single statement. This clause is used to compute aggregations
     based on multiple grouping sets. `ROLLUP` is a shorthand for `GROUPING SETS`. For example,
-    `GROUP BY warehouse, product WITH ROLLUP` is equivalent to `GROUP BY GROUPING SETS
-    ((warehouse, product), (warehouse), ())`.
+    `GROUP BY warehouse, product WITH ROLLUP` is equivalent to `GROUP BY ROLLUP(warehouse, product)` or
+    `GROUP BY GROUPING SETS((warehouse, product), (warehouse), ())`.
```

Comment:
is this corrected? I though the result is a union of `GROUP BY warehouse, product`, `GROUP BY warehouse`, `GROUP BY ()` (global aggregate).

Reply:
> is this corrected? I though the result is a union of `GROUP BY warehouse, product`, `GROUP BY warehouse`, `GROUP BY ()` (global aggregate).

```
spark-sql> CREATE TABLE dealer (id INT, city STRING, car_model STRING, quantity INT);
Time taken: 3.084 seconds
spark-sql> INSERT INTO dealer VALUES
         >     (100, 'Fremont', 'Honda Civic', 10),
         >     (100, 'Fremont', 'Honda Accord', 15),
         >     (100, 'Fremont', 'Honda CRV', 7),
         >     (200, 'Dublin', 'Honda Civic', 20),
         >     (200, 'Dublin', 'Honda Accord', 10),
         >     (200, 'Dublin', 'Honda CRV', 3),
         >     (300, 'San Jose', 'Honda Civic', 5),
         >     (300, 'San Jose', 'Honda Accord', 8);
21/03/25 15:48:47 WARN ObjectStore: Failed to get database global_temp, returning NoSuchObjectException
Time taken: 2.332 seconds
spark-sql>  select id, city, sum(quantity) from dealer group by rollup(id, city);
100	NULL	32
300	NULL	13
200	NULL	33
NULL	NULL	78
200	Dublin	33
300	San Jose	13
100	Fremont	32
Time taken: 1.388 seconds, Fetched 7 row(s)
spark-sql> select id, city, sum(quantity) from dealer group by grouping sets((id, city), (id), ());
100	NULL	32
300	NULL	13
200	NULL	33
NULL	NULL	78
200	Dublin	33
300	San Jose	13
100	Fremont	32
Time taken: 0.284 seconds, Fetched 7 row(s)
spark-sql> explain select id, city, sum(quantity) from dealer group by rollup(id, city);
== Physical Plan ==
AdaptiveSparkPlan isFinalPlan=false
+- HashAggregate(keys=[id#68, city#69, spark_grouping_id#67L], functions=[sum(quantity#62)])
   +- Exchange hashpartitioning(id#68, city#69, spark_grouping_id#67L, 200), ENSURE_REQUIREMENTS, [id=#113]
      +- HashAggregate(keys=[id#68, city#69, spark_grouping_id#67L], functions=[partial_sum(quantity#62)])
         +- Expand [List(quantity#62, id#59, city#60, 0), List(quantity#62, id#59, null, 1), List(quantity#62, null, null, 3)], [quantity#62, id#68, city#69, spark_grouping_id#67L]
            +- Scan hive default.dealer [quantity#62, id#59, city#60], HiveTableRelation [`default`.`dealer`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [id#59, city#60, car_model#61, quantity#62], Partition Cols: []]


Time taken: 0.064 seconds, Fetched 1 row(s)
spark-sql> explain select id, city, sum(quantity) from dealer group by grouping sets((id, city), (id), ());
== Physical Plan ==
AdaptiveSparkPlan isFinalPlan=false
+- HashAggregate(keys=[id#86, city#87, spark_grouping_id#85L], functions=[sum(quantity#80)])
   +- Exchange hashpartitioning(id#86, city#87, spark_grouping_id#85L, 200), ENSURE_REQUIREMENTS, [id=#136]
      +- HashAggregate(keys=[id#86, city#87, spark_grouping_id#85L], functions=[partial_sum(quantity#80)])
         +- Expand [List(quantity#80, id#77, city#78, 0), List(quantity#80, id#77, null, 1), List(quantity#80, null, null, 3)], [quantity#80, id#86, city#87, spark_grouping_id#85L]
            +- Scan hive default.dealer [quantity#80, id#77, city#78], HiveTableRelation [`default`.`dealer`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [id#77, city#78, car_model#79, quantity#80], Partition Cols: []]


Time taken: 0.063 seconds, Fetched 1 row(s)
spark-sql>
```